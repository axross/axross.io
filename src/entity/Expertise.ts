class Expertise {
  public readonly iconType: ExpertiseIconType;
  public readonly name: string;

  constructor({ iconType, name }: { iconType: ExpertiseIconType; name: string }) {
    this.iconType = iconType;
    this.name = name;
  }
}

export enum ExpertiseIconType {
  TYPESCRIPT = 'TYPESCRIPT',
  REACT = 'REACT',
  DART = 'DART',
  FLUTTER = 'FLUTTER',
  GRAPHQL = 'GRAPHQL',
  JAVASCRIPT = 'JAVASCRIPT',
  NODEJS = 'NODEJS',
  HTML = 'HTML',
  CSS = 'CSS',
}

export default Expertise;
