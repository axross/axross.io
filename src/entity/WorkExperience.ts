import { DateTime } from 'luxon';

class WorkExperience {
  public readonly startedAt: DateTime;
  public readonly endedAt: DateTime | null;
  public readonly employeeName: string;
  public readonly employeeThumbnailImageUrl: string;
  public readonly summary: string;

  public compare = (other: WorkExperience): number => {
    const thisEndedAtAsNumber = this.endedAt === null ? Number.MAX_SAFE_INTEGER : this.endedAt.valueOf();
    const otherEndedAtAsNumber = other.endedAt === null ? Number.MAX_SAFE_INTEGER : other.endedAt.valueOf();

    const comparedWithEndedAt = thisEndedAtAsNumber - otherEndedAtAsNumber;

    if (comparedWithEndedAt === 0) {
      return this.startedAt.valueOf() - other.startedAt.valueOf();
    } else {
      return comparedWithEndedAt;
    }
  };

  constructor({
    startedAt,
    endedAt,
    employeeName,
    employeeThumbnailImageUrl,
    summary,
  }: {
    startedAt: DateTime;
    endedAt: DateTime | null;
    employeeName: string;
    employeeThumbnailImageUrl: string;
    summary: string;
  }) {
    this.startedAt = startedAt;
    this.endedAt = endedAt;
    this.employeeName = employeeName;
    this.employeeThumbnailImageUrl = employeeThumbnailImageUrl;
    this.summary = summary;
  }
}

export default WorkExperience;
