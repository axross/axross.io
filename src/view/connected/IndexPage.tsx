import { createElement } from 'react';
import { Subscriber } from 'react-broadcast';
import { Dependency } from '../../dependency';
import State from '../../core/State';
import { decrement, increment } from '../../thunk/count';
import IndexPage from '../page/IndexPage';

const Connected = () => (
  <Subscriber channel="dependency">
    {({ store }: Dependency) => (
      <Subscriber channel="state">
        {(state: State) => (
          <IndexPage
            expertises={state.expertises}
            workExperiences={state.workExperiences}
            count={state.count}
            onClickIncrement={() => store.dispatch(increment())}
            onClickDecrement={() => store.dispatch(decrement())}
          />
        )}
      </Subscriber>
    )}
  </Subscriber>
);

export default Connected;
