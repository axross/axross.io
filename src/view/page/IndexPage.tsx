import glamorous from 'glamorous';
import { ClassAttributes, createElement, MouseEventHandler } from 'react';
import JobExperience from '../../entity/JobExperience';
import Counter from './Counter';

type Props = ClassAttributes<HTMLElement> & {
  jobExperiences: JobExperience[];
  count: number;
  onClickIncrement: MouseEventHandler<HTMLElement>;
  onClickDecrement: MouseEventHandler<HTMLElement>;
};

const IndexPage = ({ jobExperiences, count, onClickIncrement, onClickDecrement }: Props) => (
  <Root>
    <ul>{jobExperiences.map(jobExperience => <li key={jobExperience.employee}>{jobExperience.employee}</li>)}</ul>

    <Counter count={count} onClickIncrement={onClickIncrement} onClickDecrement={onClickDecrement} />
  </Root>
);

const Root = glamorous.div({});

export default IndexPage;
