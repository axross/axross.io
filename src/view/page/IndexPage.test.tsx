import { shallow } from 'enzyme';
import { DateTime } from 'luxon';
import { createElement } from 'react';
import Expertise from '../../entity/Expertise';
import WorkExperience from '../../entity/WorkExperience';
import IndexPage from './IndexPage';

describe('<IndexPage>', () => {
  test('it matches with the previous snapshot', () => {
    const expertises = [new Expertise({ type: 'TYPESCRIPT' }), new Expertise({ type: 'NODEJS' })];
    const introduction =
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';
    const workExperiences = [
      new WorkExperience({
        startedAt: DateTime.fromMillis(0, { locale: 'utc' }),
        endedAt: DateTime.fromMillis(0, { locale: 'utc' }),
        employeeName: 'Eighty Three, Inc.',
        employeeThumbnailImageUrl: 'https://eighty-three.co.jp/',
        summary: 'Yeah',
      }),
    ];

    const rendered = shallow(
      <IndexPage expertises={expertises} introduction={introduction} workExperiences={workExperiences} />,
    );

    expect(rendered).toMatchSnapshot();
  });
});
