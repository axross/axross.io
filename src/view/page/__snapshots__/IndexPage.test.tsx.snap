// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexPage> it matches with the previous snapshot 1`] = `
<Styled(div)>
  <Styled(Header) />
  <Styled(Introduction)>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </Styled(Introduction)>
  <Styled(Expertises)>
    <ExpertisesItem
      expertise={
        Expertise {
          "type": "TYPESCRIPT",
        }
      }
      key="TYPESCRIPT"
    />
    <ExpertisesItem
      expertise={
        Expertise {
          "type": "NODEJS",
        }
      }
      key="NODEJS"
    />
  </Styled(Expertises)>
  <Styled(Principles)>
    <PrinciplesItem
      icon={[Function]}
      name="Both of *-side"
    >
      We have to know well server-side to know client-side well, and vice versa. My specialty is client-side but I also have enough knowledge for server-side.
    </PrinciplesItem>
    <PrinciplesItem
      icon={[Function]}
      name="User Centered"
    >
      Software is made for users. I keep in mind to design UI and even to name identifiers with thinking of users.
    </PrinciplesItem>
    <PrinciplesItem
      icon={[Function]}
      name="OSS Style"
    >
      I'll create an Issue when I find some problem or some kind of improvement, I'll post an Issue, fix it or improve it, create a Pull Request, test and build on CI and then deploy it via code reviewing.
    </PrinciplesItem>
    <PrinciplesItem
      icon={[Function]}
      name="Be Agile"
    >
      I think objective feedbacks from users are more important than subjective our sense of accomplishment. I like to improve our software little by little and as agile.
    </PrinciplesItem>
    <PrinciplesItem
      icon={[Function]}
      name="As Robust"
    >
      The code which is not enough designed for application architecture will decline productivity soon. I always write robust code to keep high productivity long time.
    </PrinciplesItem>
    <PrinciplesItem
      icon={[Function]}
      name="Keep Clean"
    >
      Code ages as same as cells. But code can be refactored. I never leave code untidy.
    </PrinciplesItem>
  </Styled(Principles)>
  <Styled(JobExperiences)>
    <WorkExperiencesItem
      key="Eighty Three, Inc."
      workExperience={
        WorkExperience {
          "compare": [Function],
          "employeeName": "Eighty Three, Inc.",
          "employeeThumbnailImageUrl": "https://eighty-three.co.jp/",
          "endedAt": "1970-01-01T00:00:00.000Z",
          "startedAt": "1970-01-01T00:00:00.000Z",
          "summary": "Yeah",
        }
      }
    />
  </Styled(JobExperiences)>
  <Styled(Footer) />
</Styled(div)>
`;
